core/quantity/src/Formatter/Formatter.ts

```diff
@@ -414,10 +437,104 @@ export class Formatter {
     else
       formattedValue = formattedMagnitude;
 
-    if (spec.format.minWidth && spec.format.minWidth < formattedValue.length)
-      formattedValue.padStart(spec.format.minWidth, " ");
-
     return formattedValue;
   }
 
+  private static processBearingAndAzimuth(magnitude: number, spec: FormatterSpec): {magnitude: number, prefix?: string, suffix?: string} {
+    const type = spec.format.type;
+    if (type !== FormatType.Bearing && type !== FormatType.Azimuth)
+      return {magnitude};
+
+    const revolution = this.getRevolution(spec);
+    magnitude = this.normalizeAngle(magnitude, revolution);
+    const quarterRevolution = revolution / 4;
+
+    if (type === FormatType.Bearing) {
+      let quadrant = 0;
+      while (magnitude > quarterRevolution) {
+        magnitude -= quarterRevolution;
+        quadrant++;
+      }
+      let prefix, suffix: string;
+
+      // Quadrants are
+      // 1 0
+      // 2 3
+      // For quadrants 0 and 2 we have to subtract the angle from quarterRevolution degrees because they go clockwise
+      if (quadrant === 0 || quadrant === 2)
+        magnitude = quarterRevolution - magnitude;
+
+      // TODO: at some point we will want to open this for localization, in the first release it's going to be hard coded
+      if (quadrant === 0 || quadrant === 1)
+        prefix = "N";
+
+      if (quadrant === 2 || quadrant === 3)
+        prefix = "S";
+
+      if (quadrant === 0 || quadrant === 3)
+        suffix = "E";
+
+      if (quadrant === 1 || quadrant === 2)
+        suffix = "W";
+
+      // special case, if in quadrant 2 and value is very small, turn suffix to E because S00E is preferred over S00W
+      if (quadrant === 2 && magnitude < 0.00000000001) // 1e-11 is a small value that is close to 0
+        suffix = "E";
+
+      return {magnitude, prefix, suffix: suffix!};
+    }
+
+    if (type === FormatType.Azimuth) {
+      let azimuthBase = quarterRevolution; // default base is North
```

Comment:
Is this assuming the persisted values base is east?

Reply:
yes. North is quarter revolution, which in degrees would be 90.